activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -415,15 +415,28 @@ def apply_join_dependency(relation, join_dependency)
         end
       end
 
+      SPLIT_ORDER_VALUES_REGEX = /,(?![^()]*+\))/
```

Comment:
We avoid a lot to parse SQL in our codebase because we can't create a complete parse and the complexity is too high. If `order('foo.bar', 'foo.baz)` works I think we should raise if a complex string is passed to `order` and not increase the complexity of this code.